MREUnityRuntime/MREUnityRuntimeLib/Core/Actor.cs

```diff
@@ -774,55 +806,150 @@ private void PatchTransform(ActorTransformPatch transformPatch)
+                }
+            }
 
-                            // Transform new position to world space.
-                            transformUpdate.Position = appTransform.TransformPoint(newAppPos);
-                        }
+            if (transformPatch.App != null)
+            {
+                var appTransform = App.SceneRoot.transform;
 
-                        if (transformPatch.App.Rotation != null)
-                        {
-                            // New app space rotation
-                            var newAppRot = (transform.rotation * appTransform.rotation)
-                                .GetPatchApplied(AppTransform.Rotation.ApplyPatch(transformPatch.App.Rotation));
+                if (transformPatch.App.Position != null)
+                {
+                    // New app space position.
+                    var newAppPos = appTransform.InverseTransformPoint(transform.position)
+                        .GetPatchApplied(AppTransform.Position.ApplyPatch(transformPatch.App.Position));
 
-                            // Transform new app rotation to world space.
-                            transformUpdate.Rotation = newAppRot * transform.rotation;
-                        }
+                    // Transform new position to world space.
+                    transformUpdate.Position = appTransform.TransformPoint(newAppPos);
+                }
+
+                if (transformPatch.App.Rotation != null)
+                {
+                    // New app space rotation
+                    var newAppRot = (transform.rotation * appTransform.rotation)
+                        .GetPatchApplied(AppTransform.Rotation.ApplyPatch(transformPatch.App.Rotation));
+
+                    // Transform new app rotation to world space.
+                    transformUpdate.Rotation = newAppRot * transform.rotation;
+                }
+            }
+
+            // Queue update to happen in the fixed update
+            RigidBody.SynchronizeEngine(transformUpdate);
+        }
+
+        private void CorrectAppTransform(MWTransform transform)
+        {
+            if (transform == null)
+            {
+                return;
+            }
+
+            if (RigidBody == null)
+            {
+                // We need to lerp at the transform level with the transform lerper.
+                if (_transformLerper == null)
+                {
+                    _transformLerper = new TransformLerper(gameObject.transform);
+                }
+
+                // Convert the app relative transform for the correction to world position relative to our app root.
+                Vector3? newPos = null;
+                Quaternion? newRot = null;
+
+                if (transform.Position != null)
+                {
+                    Vector3 appPos;
+                    appPos.x = transform.Position.X;
+                    appPos.y = transform.Position.Y;
+                    appPos.z = transform.Position.Z;
```

Comment:
Are these vector patches partial? Do we need to check for nulls?

Reply:
Actor corrections do not actually work off of patches.  Actor corrections will always contain a transform with full position and/or rotation to be lerped to.  Null checks are needed around the position and rotation are needed, but if they exist, all components in the vector 3 or quaternion will exist, by design.  Thanks for calling that out though to be sure.